src/pages/Items.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import heartIcon from "../assets/ic_heart.png";
+import "../styles/items.css";
+
+import ItemTop from "../components/ItemTop";
+import Pagnation from "../components/Pagination";
+import useMediaQuery from "../utils/useQueryMedia";
+
+const filterName = {
+  recent: "최신순",
+  favorite: "좋아요순",
+};
+
+const getItems = ({ page, pageSize, orderBy, keyword }) => {
+  return fetch(
+    `https://panda-market-api.vercel.app/products?page=${page}&pageSize=${pageSize}&orderBy=${orderBy}&keyword=${keyword}`
+  ).then((response) => response.json());
+};
+
+function Items() {
+  const isMobile = useMediaQuery("(max-width: 767px)");
+  const isTablet = useMediaQuery("(max-width: 1199px)");
+  const [bests, setBests] = useState([]);
+  const [sells, setSells] = useState([]);
+  const [total, setTotal] = useState(0);
+  const [page, setPage] = useState(1);
+  const [filter, setFilter] = useState("recent");
+
+  useEffect(() => {
+    async function getFavoriteProducts() {
+      const data = await getItems({
+        page: 1,
+        pageSize: 4,
+        orderBy: "favorite",
+        keyword: 0,
+      });
+      setBests(data.list);
+    }
+    getFavoriteProducts();
+  }, []);
+
+  useEffect(() => {
+    async function getRecentProducts() {
+      const data = await getItems({
+        page: page,
+        pageSize: isMobile ? 4 : isTablet ? 6 : 10,
+        orderBy: filter,
+        keyword: 0,
+      });
+      setSells(data.list);
+      setTotal(data.totalCount);
+    }
+    getRecentProducts();
+  }, [filter, page, isMobile, isTablet]);
```

Comment:
중첩된 삼항연산자도 가능하지만
가독성이 좋은건 아무래도 조건문이 가독성이 좋긴 합니다.

```suggestion
const displayDefault = (isMobile,isTablet)=>{
if(isTablet) return 4
if(isMobile) return 6
return 10;
}

  useEffect(() => {
  const pageSize = displayDefault(isMobile,isTablet);
    async function getRecentProducts() {
      const data = await getItems({
        page: page,
        pageSize: pageSize,
        orderBy: filter,
        keyword: 0,
      });
      setSells(data.list);
      setTotal(data.totalCount);
    }
    getRecentProducts();
  }, [filter, page, isMobile, isTablet]);
```

displayDefault를 조건문으로 처리하는 함수를 만들어서 반환된 결과로 pageAmount를 설정하는 방법도 가능할것같아요

Reply:
맞아요. 시간이 촉박해서 못했는데, 많이 사용되서 수정해야겠다 생각했습니다. 이런 방식으로 하니 깔끔하군요